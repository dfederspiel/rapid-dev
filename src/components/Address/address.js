import React from 'react';
import { Column, Row } from "simple-flexbox";

var stateList = ['AK', 'AL', 'AR', 'AZ', 'CA', 'CO', 'CT', 'DC', 'DE', 'FL', 'GA', 'HI', 'IA', 'ID', 'IL', 'IN', 'KS', 'KY', 'LA', 'MA', 'MD', 'ME', 'MI', 'MN', 'MO', 'MS', 'MT', 'NC', 'ND', 'NE', 'NH', 'NJ', 'NM', 'NV', 'NY', 'OH', 'OK', 'OR', 'PA', 'PR', 'RI', 'SC', 'SD', 'TN', 'TX', 'UT', 'VA', 'VT', 'WA', 'WI', 'WV', 'WY'];
var addressTypes = [{ name: 'Corporate Office', value: 1 }, { name: 'Branch Office', value: 2 }, { name: 'Home', value: 3 }, { name: 'Business', value: 4 }];

export default class Address extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            type: props.type,
            address1: props.address1,
            address2: props.address2,
            address3: props.address3,
            city: props.city,
            state: props.state,
            zip: props.zip,
            zip4: props.zip4,
            errorType: false,
            errorAddress1: false,
            errorCity: false,
            errorZip: false,
            errorState: false
        };
    }

    handleInputChange= (event) => {
        const target = event.target;
        const value = target.value;
        const name = target.id;

        this.setState({
            [name]: value
        }, () => { this.validate(); });
    }

    validate = () => {
        this.setState({
            errorType: this.state.type.length === 0,
            errorAddress1: this.state.address1.length === 0,
            errorCity: this.state.city.length === 0,
            errorZip: this.state.zip.length === 0,
            errorState: this.state.state.length === 0
        });
    }

render(){
        return (
            <Column flexGrow={1} className="flex-column">
                <Row>
                    <Column>
                        <label htmlFor="type-select">Type*</label>
                        <select id="type-select" className={this.state.errorType ? "error" : ""} defaultValue={this.state.type} onChange={this.handleInputChange}>
                            {
                                addressTypes.map((type) => {
                                    return <option key={type.value} value={type.value}>{type.name}</option>;
                                })
                            }
                        </select>
                        {
                            this.state.errorType &&
                            <label><span className="error">*</span> Required Field</label>
                        }
                    </Column>
                </Row>
                <Row>
                    <Column>
                        <label htmlFor="address1">Address 1*</label>
                        <input type="text" id="address1" value={this.state.address1} className={this.state.errorAddress1 ? "error" : ""} onChange={this.handleInputChange} />
                        {
                            this.state.errorAddress1 &&
                            <label><span className="error">*</span> Required Field</label>
                        }
                    </Column>
                </Row>
                <Row>
                    <Column>
                        <label htmlFor="address2">Address 2</label>
                        <input type="text" id="address2" value={this.state.address2} onChange={this.handleInputChange}/>
                    </Column>
                </Row>
                <Row>
                    <Column>
                        <label htmlFor="address3">Address 3</label>
                        <input type="text" id="address3" value={this.state.address3} onChange={this.handleInputChange}/>
                    </Column>
                </Row>
                <Row wrap={true}>
                    <Column className="spaced-element">
                        <label htmlFor="city">City*</label>
                        <input type="text" id="city" value={this.state.city} className={this.state.errorCity ? "error" : ""} onChange={this.handleInputChange} />
                        {
                            this.state.errorCity &&
                            <label><span className="error">*</span> Required Field</label>
                        }
                    </Column>
                    <Column>
                        <label htmlFor="state">State*</label>
                        <select id="state" className={this.state.errorState ? "error" : ""} value={this.state.state} onChange={this.handleInputChange}>
                            {
                                stateList.map((state)=> {
                                    return <option key={state} value={state}>{state}</option>;
                                })
                            }
                        </select>
                        {
                            this.state.errorState &&
                            <label><span className="error">*</span> Required Field</label>
                        }
                    </Column>
                </Row>
                <Row>
                    <Column>
                        <Row>
                            <Column>
                                <label htmlFor="zip">Zip Code*</label>
                            </Column>
                        </Row>
                        <Row wrap={true}>
                            <Column className="spaced-element">
                                <input type="text" id="zip" value={this.state.zip} className={this.state.errorZip ? "error" : ""} onChange={this.handleInputChange} />
                                {
                                    this.state.errorZip &&
                                    <label><span className="error">*</span> Required Field</label>
                                }
                            </Column>
                            <Column>
                                <input type="text" id="zip4" value={this.state.zip4} onChange={this.handleInputChange}/>
                            </Column>
                        </Row>
                    </Column>
                </Row>
            </Column>
        );
    }
}