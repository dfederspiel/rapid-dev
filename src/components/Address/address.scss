.flex-column > div{
    margin-bottom:1.5rem;
}

#address1,
#address2,
#address3{
    width:40rem;
}

#zip4{
    width:10rem;
}