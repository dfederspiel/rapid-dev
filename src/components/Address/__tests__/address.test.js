import React from 'react';
import Address from '../address';
import renderer from 'react-test-renderer';

describe('the address component', () => {

    const props = {
        type: 2,
        address1: '100 N Main',
        address2: 'PO Box 123',
        address3: 'Apt 999',
        city: 'Lima',
        state: 'OH',
        zip: '45801',
        zip4: '1234'
    };

    it('renders correctly', () => {
        const tree = renderer.create(
            <Address
                {...props}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});