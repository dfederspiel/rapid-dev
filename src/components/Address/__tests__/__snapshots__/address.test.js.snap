// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the address component renders correctly 1`] = `
<div
  className="flex-column"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "flexGrow": 1,
      "flexWrap": "nowrap",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <label
        htmlFor="type-select"
      >
        Type*
      </label>
      <select
        className=""
        defaultValue={2}
        id="type-select"
        onChange={[Function]}
      >
        <option
          value={1}
        >
          Corporate Office
        </option>
        <option
          value={2}
        >
          Branch Office
        </option>
        <option
          value={3}
        >
          Home
        </option>
        <option
          value={4}
        >
          Business
        </option>
      </select>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <label
        htmlFor="address1"
      >
        Address 1*
      </label>
      <input
        className=""
        id="address1"
        onChange={[Function]}
        type="text"
        value="100 N Main"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <label
        htmlFor="address2"
      >
        Address 2
      </label>
      <input
        id="address2"
        onChange={[Function]}
        type="text"
        value="PO Box 123"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <label
        htmlFor="address3"
      >
        Address 3
      </label>
      <input
        id="address3"
        onChange={[Function]}
        type="text"
        value="Apt 999"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <div
      className="spaced-element"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <label
        htmlFor="city"
      >
        City*
      </label>
      <input
        className=""
        id="city"
        onChange={[Function]}
        type="text"
        value="Lima"
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <label
        htmlFor="state"
      >
        State*
      </label>
      <select
        className=""
        id="state"
        onChange={[Function]}
        value="OH"
      >
        <option
          value="AK"
        >
          AK
        </option>
        <option
          value="AL"
        >
          AL
        </option>
        <option
          value="AR"
        >
          AR
        </option>
        <option
          value="AZ"
        >
          AZ
        </option>
        <option
          value="CA"
        >
          CA
        </option>
        <option
          value="CO"
        >
          CO
        </option>
        <option
          value="CT"
        >
          CT
        </option>
        <option
          value="DC"
        >
          DC
        </option>
        <option
          value="DE"
        >
          DE
        </option>
        <option
          value="FL"
        >
          FL
        </option>
        <option
          value="GA"
        >
          GA
        </option>
        <option
          value="HI"
        >
          HI
        </option>
        <option
          value="IA"
        >
          IA
        </option>
        <option
          value="ID"
        >
          ID
        </option>
        <option
          value="IL"
        >
          IL
        </option>
        <option
          value="IN"
        >
          IN
        </option>
        <option
          value="KS"
        >
          KS
        </option>
        <option
          value="KY"
        >
          KY
        </option>
        <option
          value="LA"
        >
          LA
        </option>
        <option
          value="MA"
        >
          MA
        </option>
        <option
          value="MD"
        >
          MD
        </option>
        <option
          value="ME"
        >
          ME
        </option>
        <option
          value="MI"
        >
          MI
        </option>
        <option
          value="MN"
        >
          MN
        </option>
        <option
          value="MO"
        >
          MO
        </option>
        <option
          value="MS"
        >
          MS
        </option>
        <option
          value="MT"
        >
          MT
        </option>
        <option
          value="NC"
        >
          NC
        </option>
        <option
          value="ND"
        >
          ND
        </option>
        <option
          value="NE"
        >
          NE
        </option>
        <option
          value="NH"
        >
          NH
        </option>
        <option
          value="NJ"
        >
          NJ
        </option>
        <option
          value="NM"
        >
          NM
        </option>
        <option
          value="NV"
        >
          NV
        </option>
        <option
          value="NY"
        >
          NY
        </option>
        <option
          value="OH"
        >
          OH
        </option>
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="OR"
        >
          OR
        </option>
        <option
          value="PA"
        >
          PA
        </option>
        <option
          value="PR"
        >
          PR
        </option>
        <option
          value="RI"
        >
          RI
        </option>
        <option
          value="SC"
        >
          SC
        </option>
        <option
          value="SD"
        >
          SD
        </option>
        <option
          value="TN"
        >
          TN
        </option>
        <option
          value="TX"
        >
          TX
        </option>
        <option
          value="UT"
        >
          UT
        </option>
        <option
          value="VA"
        >
          VA
        </option>
        <option
          value="VT"
        >
          VT
        </option>
        <option
          value="WA"
        >
          WA
        </option>
        <option
          value="WI"
        >
          WI
        </option>
        <option
          value="WV"
        >
          WV
        </option>
        <option
          value="WY"
        >
          WY
        </option>
      </select>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "nowrap",
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "flexWrap": "nowrap",
            }
          }
        >
          <label
            htmlFor="zip"
          >
            Zip Code*
          </label>
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <div
          className="spaced-element"
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "flexWrap": "nowrap",
            }
          }
        >
          <input
            className=""
            id="zip"
            onChange={[Function]}
            type="text"
            value="45801"
          />
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "flexWrap": "nowrap",
            }
          }
        >
          <input
            id="zip4"
            onChange={[Function]}
            type="text"
            value="1234"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
